src/js/utils/debug.js

```diff
@@ -17,51 +13,56 @@ const colors: {[k: string]: string} = {
 type LogMessage = {
     level: string;
     prefix: string;
-    message: Array<any>;
+    message: any[];
     timestamp: number;
 }
 
-export default class Log {
+const MAX_ENTRIES = 100;
+
+class Log {
     prefix: string;
     enabled: boolean;
     css: string;
-    messages: Array<LogMessage>;
+    messages: LogMessage[];
 
-    constructor(prefix: string, enabled: boolean = false) {
+    constructor(prefix: string, enabled: boolean) {
         this.prefix = prefix;
         this.enabled = enabled;
         this.messages = [];
         this.css = colors[prefix] || 'color: #000000; background: #FFFFFF;';
     }
 
-    addMessage(level: string, prefix: string, ...args: Array<any>): void {
+    addMessage(level: string, prefix: string, ...args: any[]) {
         this.messages.push({
             level: level,
             prefix: prefix,
             message: args,
             timestamp: new Date().getTime(),
         });
+        if (this.messages.length > MAX_ENTRIES) {
+            this.messages.pop();
```

Comment:
Maybe I am missing something, but this way only first <MAX_ENTRIES> will be logged, each message after hitting this limit will be pushed at the end of the array and then immediately popped from it.

Reply:
nice! good catch. i've mindlessly copy/pasted it [from suite](https://github.com/trezor/trezor-suite/blob/develop/packages/suite/src/reducers/suite/logReducer.ts#L38) :)

Reply:
addresses in suite as well https://github.com/trezor/trezor-suite/pull/3494